docs/BreakingChanges/054 List.Sort algorithm changed.md

```diff
@@ -0,0 +1,25 @@
+## 54: List.Sort algorithm changed
```

Comment:
{
  "Id": "54",
  "Title": "List.Sort algorithm changed",
  "Details": "Begining in .NET Framework 4.5, `List<T>`'s sort algorithm has changed (to be an introspective sort instead of a quick sort). `List<T>`'s sort has never been stable, but this change may cause different scenarios to sort in unstable ways. That simply means that equivalent items may sort in different orders in subsequent calls of the API.",
  "Suggestion": "Because the old sort algorithm was also unstable (though in slightly different ways), there should be no code that depends on equivalent items always sorting in a particular order. If there are instances of code depending upon that and being lucky with the old behavior, that code should be updated to use a comparer that will deterministically sort the items in the desired order.",
  "Link": "http://stackoverflow.com/questions/12461328/behaviour-of-listt-sort-in-net-4-5-changed-from-net-4-0?rq=1",
  "Markdown": null,
  "ApplicableApis": [
    "M:System.Collections.Generic.List`1.Sort",
    "M:System.Collections.Generic.List`1.Sort(System.Collections.Generic.IComparer{`0})",
    "M:System.Collections.Generic.List`1.Sort(System.Comparison{`0})",
    "M:System.Collections.Generic.List`1.Sort(System.Int32,System.Int32,System.Collections.Generic.IComparer{`0})"
  ],
  "Related": null,
  "VersionBroken": "4.5",
  "VersionFixed": null,
  "IsBuildTime": false,
  "IsQuirked": false,
  "IsSourceAnalyzerAvailable": false,
  "BugLink": null,
  "Notes": null,
  "IsRetargeting": false,
  "ImpactScope": 4
}

**Parsed APIs:**
- [x] M:System.Collections.Generic.List`1.Sort
- [x] M:System.Collections.Generic.List`1.Sort(System.Collections.Generic.IComparer{`0})
- [x] M:System.Collections.Generic.List`1.Sort(System.Comparison{`0})
- [x] M:System.Collections.Generic.List`1.Sort(System.Int32,System.Int32,System.Collections.Generic.IComparer{`0})